controllers/logging/logr.go

```diff
@@ -76,22 +76,38 @@ func (a *logrSinkAdapter) Error(err error, msg string, keysAndValues ...interfac
 }
 
 func (a *logrSinkAdapter) WithValues(keysAndValues ...interface{}) logr.LogSink {
+	newKV := make(map[string]string)
+	for k, v := range a.keysAndValues {
+		newKV[k] = v
+	}
 	for i := 0; i < len(keysAndValues)/2; i++ {
 		keyIndex := i * 2
 		valIndex := keyIndex + 1
-		key := fmt.Sprintf("%s", keysAndValues[keyIndex])
-		if i*2+1 > len(keysAndValues) {
+		if valIndex >= len(keysAndValues) {
 			continue
 		}
-		value := fmt.Sprintf("%s", keysAndValues[valIndex])
-		a.keysAndValues[key] = value
+		key := fmt.Sprintf("%v", keysAndValues[keyIndex])
+		value := fmt.Sprintf("%v", keysAndValues[valIndex])
+		newKV[key] = value
+	}
+	return &logrSinkAdapter{
+		z:             a.z,
+		keysAndValues: newKV,
+		name:          a.name,
 	}
-	return a
 }
 
 func (a *logrSinkAdapter) WithName(name string) logr.LogSink {
-	a.name = name
-	return a
+	// Vytvoř nový adapter s kopií hodnot
```

Comment:
it's lovely to see Czech in the codebase, but can we translate this comment to English for the international community :)